packages/x-date-pickers/src/internals/utils/date-utils.ts

```diff
@@ -203,6 +203,7 @@ export const getWeekdays = <TDate extends PickerValidDate>(
   utils: MuiPickersAdapter<TDate>,
   date: TDate,
 ) => {
-  const start = utils.startOfWeek(date);
+  const cleanDate = utils.setLocaleToValue(date);
```

Comment:
The alternative solution would be to do `utils.` inside `utils.startOfWeek` and remove it's usage in `utils.getWeekArray` and here.

Advantages:
- No new public method
- No impact on Date Fns adapter which does not care about this

Disavantages:
- Could this be needed for other methods than `utils.startOfWeek`? In which case setting the locale explicitly before processing the value is probably simpler than doing it in tons of adapter methods
- Small overhead on `utils.startOfWeek`